plugins/org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/generator/EclipseBasedShouldGenerate.xtend

```diff
@@ -0,0 +1,51 @@
+/*******************************************************************************
+ * Copyright (c) 2015 itemis AG (http://www.itemis.eu) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+package org.eclipse.xtext.ui.generator
+
+import org.eclipse.xtext.generator.IShouldGenerate
+import org.eclipse.emf.ecore.resource.Resource
+import org.eclipse.xtext.util.CancelIndicator
+import com.google.inject.Inject
+import org.eclipse.core.resources.IWorkspaceRoot
+import org.eclipse.xtext.util.Pair
+import org.eclipse.core.resources.IProject
+import org.eclipse.core.resources.IResource
+import org.eclipse.core.resources.IFile
+import org.eclipse.core.resources.IStorage
+import org.eclipse.core.runtime.CoreException
+import org.eclipse.emf.common.util.WrappedException
+import org.eclipse.core.resources.IMarker
+import org.eclipse.xtext.ui.resource.IStorage2UriMapper
+
+/**
+ * @author Sven Efftinge - Initial contribution and API
+ */
+class EclipseBasedShouldGenerate implements IShouldGenerate {
+
+	@Inject IWorkspaceRoot workspaceRoot
+	@Inject IStorage2UriMapper storage2UriMapper
+
+	override shouldGenerate(Resource resource, CancelIndicator cancelIndicator) {
```

Comment:
This will build other resources from the resource set, too since the project context is not passed from the builder to this method? I'd think IShouldGenerate should also take the current project configuration?